lib/identity_cache/cache_key_generation.rb

```diff
@@ -35,12 +35,12 @@ def prefixed_rails_cache_key
         "#{rails_cache_key_namespace}blob:#{base_class.name}:#{rails_cache_key_prefix}:"
       end
 
-      def rails_cache_index_key_for_fields_and_values(fields, values)
-        "#{rails_cache_key_namespace}index:#{base_class.name}:#{rails_cache_string_for_fields_and_values(fields, values)}"
-      end
-
-      def rails_cache_key_for_attribute_and_fields_and_values(attribute, fields, values)
-        "#{rails_cache_key_namespace}attribute:#{base_class.name}:#{attribute}:#{rails_cache_string_for_fields_and_values(fields, values)}"
+      def rails_cache_key_for_attribute_and_fields_and_values(attribute, fields, values, unique)
+        "#{rails_cache_key_namespace}" \
+          "attr#{unique ? '' : 's' }" \
```

Comment:
nitpick: I'd put the `unique ? '' : 's' ` outside of the interpolation for better readability.